### PR TITLE
[t115272] Rules for operating units regarding stock.move.lines are taken from the associated stock.move.

### PR DESCRIPTION
**Pull request by BT-cserra**
**User Stories:** mt1739

**Notes for testing/deployment:** Update manually (if not done automatically) the module ```stock_operating_unit```.

 
<div> Links to Odoo: </div> <ul>
<li><a target="_blank" href="https://odoo.braintec-group.com/web#view_type=form&model=project.task&id=115272">[t115272] [MIUS] [DEV] [mt1739] USA: Mifroma - Inventory by Lot Report to Show Case Landed Costs for Case Items</a></li>
</ul>
<div>Affected Modules:</div>
<table><thead><tr><th>Module</th><th>Ext</th></tr></thead>
<tr><td>stock_operating_unit</td><td>.py, .xml</td></tr>
<tr><td>product_operating_unit</td><td>.py, .xml</td></tr>
</tbody></table>

### CODE REVIEW

**Review by BT-rmartin — APPROVED** [review]

File: product_operating_unit/tests/test_product_operating_unit.py [hunk 1]
```diff
@@ -17,7 +17,8 @@ def setUp(self):
         # groups
         self.group_stock_user = self.env.ref('stock.group_stock_user')
         # Main Operating Unit
-        self.ou1 = self.env.ref('operating_unit.main_operating_unit')
```
Comment (BT-rmartin) [inline product_operating_unit/tests/test_product_operating_unit.py:20]: Why can't we use main_operating_unit when this module depends on operating_unit and it's defined there?

File: product_operating_unit/__manifest__.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "data": [
         "security/product_template_security.xml",
         "views/product_template_view.xml",
+        "data/product_operating_unit_data.xml",
```
Comment (BT-rmartin) [inline product_operating_unit/__manifest__.py:21]: I don't find it in the guidelines, but I think the order tends to be data, views, security

File: stock_operating_unit/security/stock_security.xml [hunk 3]
```diff
@@ -118,4 +118,20 @@
         <field eval="0" name="perm_create"/>
     </record>
 
+    <record id="ir_rule_stock_move_line_allowed_operating_units" model="ir.rule">
+        <field name="model_id" ref="stock.model_stock_move_line"/>
+        <field name="domain_force">['|',
```
Comment (BT-rmartin) [inline stock_operating_unit/security/stock_security.xml:123]: This domain is (A|B) & (C|D)? If so it's ok

File: stock_operating_unit/tests/test_stock_operating_unit.py [hunk 4]
```diff
@@ -19,10 +21,20 @@ def setUp(self):
...
-        self.product1 = self.env.ref('product.product_product_7')
-        self.product2 = self.env.ref('product.product_product_9')
-        self.product3 = self.env.ref('product.product_product_11')
+
+        # Products. Other modules set an operating unit on the product
+        # taking it from the operating unit of the user. We don't want to
+        # depend on those other modules, thus we work-around that feature
+        # by setting (for a moment) the default operating unit of the user
+        # to be empty while creating the product.
+        self.product_categ = self.ProductCategObj.create({
```
Comment (BT-rmartin) [inline stock_operating_unit/tests/test_stock_operating_unit.py:30]: I see this workaround, but why it was a problem creating a product with the operating unit of the user?
That the user running the tests cannot access to it?

**Review by BT-rmartin — CHANGES_REQUESTED** [review]
Pleas, check my comments. Probably no change is necessary

File: product_operating_unit/tests/test_product_operating_unit.py (hunk 1 above)
Comment (BT-cserra) [reply product_operating_unit/tests/test_product_operating_unit.py:20]: We can use it, but it complicates making our tests run when other modules (which also use that OU) are installed. By using our own demo data we control which OU we use, and we are more independent from other modules.

File: stock_operating_unit/security/stock_security.xml (hunk 3 above)
Comment (BT-cserra) [reply stock_operating_unit/security/stock_security.xml:123]: Yes.

File: stock_operating_unit/tests/test_stock_operating_unit.py (hunk 4 above)
Comment (BT-cserra) [reply stock_operating_unit/tests/test_stock_operating_unit.py:30]: This test assumes certain operating units (OUs) are on certain products. When ```product_operating_unit``` was installed, that assumption was broken and the test failed.

File: product_operating_unit/__manifest__.py (hunk 2 above)
Comment (BT-cserra) [reply product_operating_unit/__manifest__.py:21]: OK ─ I think the order you mention is the usual one, yes.

File: product_operating_unit/tests/test_product_operating_unit.py (hunk 1 above)
Comment (BT-rmartin) [reply product_operating_unit/tests/test_product_operating_unit.py:20]: Ok, then leave it more independent ;-)

**Pull request closed without merging**